056-kafka-roller-kraft.md

```diff
@@ -0,0 +1,131 @@
+# Kafka Roller KRaft Support
+
+This proposal describes the actions that the KafkaRoller should take when operating 
+against a Strimzi cluster in KRaft mode.
+The proposal omits details about exactly how this should be implemented because 
+the logic of the KafkaRoller is currently being rewritten.
+This proposal only describes the expected behaviour so that it can be implemented 
+in either the current KafkaRoller or a future iteration of the KafkaRoller.
+
+This proposal assumes that liveness/readiness of nodes is as described in [proposal #46](https://github.com/strimzi/proposals/blob/main/046-kraft-liveness-readiness.md) 
+and [PR #8892](https://github.com/strimzi/strimzi-kafka-operator/pull/8892).
+
+
+## Current situation
+
+When operating on a ZooKeeper based cluster the KafkaRoller has the behaviour 
+described below.
+
+### Order to roll nodes
+Roll in order:
+ - Unready brokers
+ - Ready brokers
+ - Current controller
+
+### Triggers
+The following are some of the triggers that roll a broker:
+- Pod is annotated for manual rolling update
+- Pod is unready because it is in `CrashLoopBackOff`, `ImagePullBackOff`, `ContainerCreating` or `Pending` and `Unschedulable`, 
```

Comment:
With ContainerCrating and Pending I think you mean "stuck in that state". Maybe we can create a dedicated bullet point for them.